src/pages/DataAccessRequestApplication_new.js

```diff
@@ -368,7 +362,7 @@ const DataAccessRequestApplicationNew = (props) => {
     formattedFormData.userId = userId;
 
     try {
-      let referenceId = formattedFormData.referenceId;
+      let referenceId = formData.referenceId;
```

Comment:
`formattedFormData` is a `cloneDeep` of `formData`, but didn't contain the `referenceId` value after the clone.